### ISSUE TITLE
Style assistant crashes

### ISSUE THREAD

**Issue by OCapel** [qgis/QGIS#37650]
## User Feedback

Every time when changing the style of a dataset to a more continuous color scale using the Style Assistant, QGIS crashes. It happens at the moment of applying the steps

## Report Details

**Crash ID**: [f2820dd37f18dd74add891e20920c8a7670c5673](https://github.com/qgis/QGIS/search?q=f2820dd37f18dd74add891e20920c8a7670c5673&type=Issues)

**QGIS Info**
QGIS Version: 3.14.0-Pi
QGIS code revision: 9f7028fd23
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18362

**Describe the bug**
I was editing the fill-colors of a layer to a more continuous scale, using the style assistant, and after setting the settings (i.e. define source, min and max value of the dataset, and the color ramp), QGIS shows the bugsplat "QGIS unexpectedly ended" with the options to create a bug report, quit the program, or reload the program.

**How to Reproduce**
1. Go to Layer Styling of a layer.
2. Go to the symbol settings, in my case I want to set the fill color of a simple fill.
3. Click on the icon right of 'Fill color'-bar, and click on Assistant.
4. Select a source, min and max value, and select color ramp 'Inferno'

**QGIS and OS versions**

QGIS version | 3.14.0-Pi | QGIS code revision | 9f7028fd23
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | Windows 10 (10.0)
Active python plugins | ede_interpolation; mmqgis; openlayers_plugin; db_manager; MetaSearch; processing

**Additional context**

**Comment by gioman** [comment]
@OCapel no crash here with the same steps on QGIS master on Linux.

Would you mind trying with a new QGIS profile? If the problem persist please attach a sample project+data.

**Comment by OCapel** [comment]
@gioman thank you for your response.
As I was trying to obtain the same results with a new QGIS profile, I encountered some other problems with loading the WFS (which I have created myself and for now support only locally).
In the meantime, I figured out that I was somewhat premature because if I wait a little longer between each step, it works fine. So I think that it is due to the size of my dataset.
